### ISSUE TITLE
PlanChangePolicy changing subscription period instead of price

### ISSUE THREAD

**Issue by shnela** [swappsco/django-plans#8]
Now we can charge user for changing plan when he wants to upgrade it, but when user choose cheaper option all we can do is allowing him doing it for free. It's unfair.
Here should be option for changing UserPlan.expire when changing plan.

**Comment by cdvv7788** [comment]
This feature should be assuming plans basic (10USD) and premium (20USD):
basic -> premium: Calculate the price difference, charge the user and change plan. The end expiration date should remain the same, the amount paid was only for the plan change.
premium -> basic: Calculate how much time the user should have with the remaining money in the cheaper plan and change plan and move expiration date into the future according to that.

@shnela If i understand you correctly, the second condition is not working. Am i right?

**Comment by rdlabspl** [comment]
@cdvv7788 yes, there's problem with second option (assuming that we want to be fair with our clients).

I think that there should be two kinds of strategy. First, which calculates charge for changing plan, and second which will calculate new expiration date when user chose cheaper option.
